draft-ietf-quic-tls.md

```diff
@@ -696,56 +695,50 @@ for its PRF.
+         HKDF-Expand(Secret, HkdfLabel, Length)
+
+    Where HkdfLabel is specified as:
+
+    struct {
+        uint16 length = Length;
+        opaque label<10..255> = "TLS 1.3, " + Label;
+        uint8 hashLength;     // Always 0
+    } HkdfLabel;
+~~~
+
+For example, the client packet protection secret uses an info parameter of:
 
 ~~~
    info = (HashLen / 256) || (HashLen % 256) || 0x21 ||
           "TLS 1.3, QUIC client 1-RTT secret" || 0x00
 ~~~
 
-
 ### Packet Protection Key and IV
 
 The complete key expansion uses an identical process for key expansion as
 defined in Section 7.3 of {{!I-D.ietf-tls-tls13}}, using different values for
 the input secret.  QUIC uses the AEAD function negotiated by TLS.
 
 The packet protection key and IV used to protect the 0-RTT packets sent by a
-client use the QUIC 0-RTT secret.  This uses the HKDF-Expand-Label with the PRF
-hash function negotiated by TLS.
-
-The length of the output is determined by the requirements of the AEAD function
-selected by TLS.  The key length is the AEAD key size.  As defined in Section
-5.3 of {{!I-D.ietf-tls-tls13}}, the IV length is the larger of 8 or N_MIN (see
-Section 4 of {{!RFC5116}}).
-
-~~~
-   client_0rtt_key = HKDF-Expand-Label(client_0rtt_secret,
-                                       "key", "", key_length)
-   client_0rtt_iv = HKDF-Expand-Label(client_0rtt_secret,
-                                      "iv", "", iv_length)
-~~~
-
-Similarly, the packet protection key and IV used to protect 1-RTT packets sent
-by both client and server use the current packet protection secret.
+client are derived from the QUIC 0-RTT secret. The packet protection keys
+and IVs for 1-RTT packets sent by the client and server are derived from
+the current generation of client_pp_secret and server_pp_secret respectively,
+as shown below:
 
 ~~~
-   client_pp_key_<N> = HKDF-Expand-Label(client_pp_secret_<N>,
-                                         "key", "", key_length)
-   client_pp_iv_<N> = HKDF-Expand-Label(client_pp_secret_<N>,
-                                        "iv", "", iv_length)
-   server_pp_key_<N> = HKDF-Expand-Label(server_pp_secret_<N>,
-                                         "key", "", key_length)
-   server_pp_iv_<N> = HKDF-Expand-Label(server_pp_secret_<N>,
-                                        "iv", "", iv_length)
+   key = HKDF-Expand-Label(S, "key", "", key_length)
+   iv  = HKDF-Expand-Label(S, "iv", "", iv_length)
 ~~~
 
-The client protects (or encrypts) packets with the client packet protection key
-and IV; the server protects packets with the server packet protection key.
+The length of the output is determined by the requirements of the AEAD function
+selected by TLS.  The key length is the AEAD key size.  As defined in Section
+5.3 of {{!I-D.ietf-tls-tls13}}, the IV length is the larger of 8 or N_MIN (see
+Section 4 of {{!RFC5116}}). For any secret S, the corresponding key and
+IV are derived as shown below:
```

Comment:
There is an above, but not below.

I think that you mean to move the above code block to below this paragraph and remove the "as shown below:" from the previous one.